### ISSUE TITLE
Trying to update an already up-to-date cluster breaks it

### ISSUE THREAD

**Issue by quentez** [flynn/flynn#2851]
I tried running `sudo flynn-host update` on a freshly updated cluster, and I got the following error:

```
INFO[05-19|21:21:19] pulling binaries and config              host=ip172311695
INFO[05-19|21:21:19] pulling binaries and config              host=ip172318138
INFO[05-19|21:21:22] binaries and config pulled successfully  host=ip1723131163
INFO[05-19|21:21:23] binaries and config pulled successfully  host=ip172312166
INFO[05-19|21:21:23] binaries and config pulled successfully  host=ip172311695
INFO[05-19|21:21:23] binaries and config pulled successfully  host=ip17231228
INFO[05-19|21:21:24] binaries and config pulled successfully  host=ip172318138
INFO[05-19|21:21:24] validating binaries
INFO[05-19|21:21:24] updating flynn-host daemon on all hosts
INFO[05-19|21:21:24] flynn-host updated successfully          host=ip172318138
INFO[05-19|21:21:24] flynn-host updated successfully          host=ip1723131163
INFO[05-19|21:21:24] flynn-host updated successfully          host=ip17231228
INFO[05-19|21:21:24] flynn-host updated successfully          host=ip172312166
INFO[05-19|21:21:24] flynn-host updated successfully          host=ip172311695
EROR[05-19|21:21:25] error getting cluster status             err="Get http://status-web.discoverd: dial tcp: lookup status-web.discoverd: no such host"
21:21:26.423601 host.go:138: exec: job exited with status 1
```

The cluster is now completely down and unresponsive. [I collected the logs here](https://gist.github.com/f12be8565af53980aa9315cef56a0d5b).

(This is the 3rd time it happens to me this week, with different versions of Flynn, so I don't think it's an isolated problem. On the bright side, the in-place update has started to work for me this week as well).

**Comment by hadifarnoud** [comment]
I had same issue. how can I fix it? doing a backup/restore failed on `05:04:17.687000 status-check status error: bootstrap: timed out waiting for http://status-web.discoverd, last response Get http://status-web.discoverd: dial tcp: lookup status-web.discoverd on 8.8.8.8:53: no such host`

I have same issue as #2802 after trying to restore the backup.

**Comment by lmars** [comment]
FYI I believe this was fixed during the migration to libcontainer (#3030, specifically [this change](https://github.com/flynn/flynn/pull/3030/commits/f754144c7c39d7e3289d83ff8659a75cc55aa660)). Previously every update would "resurrect" another discoverd job even though one was already running, causing issues.

If you continue to have the discoverd issue with in-place updates on releases since `v20160711.0` then let us know.
